src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -414,14 +418,14 @@ public override ImmutableArray<Location> Locations
 
         internal sealed override SyntaxNode GetDeclaratorSyntax()
         {
-            return _identifierToken.Parent;
+            return _identifierToken.Parent!;
         }
 
         public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
         {
             get
             {
-                SyntaxNode node = _identifierToken.Parent;
+                SyntaxNode node = _identifierToken.Parent!;
```

Comment:
>_identifierToken.Parent! [](start = 34, length = 24)

How do we know? (also applies a couple of lines above)

Reply:
It seems to be an undocumented condition for this type. Open to ideas for improving it.